spec/rubocop/cop/performance/double_start_end_with_spec.rb

```diff
@@ -0,0 +1,89 @@
+# encoding: utf-8
+
+require 'spec_helper'
+
+describe RuboCop::Cop::Performance::DoubleStartEndWith do
+  subject(:cop) { described_class.new }
+
+  before do
+    inspect_source(cop, source)
+  end
+
+  context 'two #start_with? calls' do
+    context 'with the same receiver' do
+      context 'all parameters of the second call are pure' do
+        let(:source) { 'x.start_with?(a, b) || x.start_with?("c", D)' }
```

Comment:
Well, strictly speaking we can't allow local variables and constants too. There might be some pathalogical cases when `.to_str` have side effects:

```ruby
class Evil
  def initialize
    @state = 0
  end

  def to_str
    @state = (@state + 1) % 4
    @state.to_s
  end
end

EVIL = Evil.new

p "1".start_with?(EVIL, EVIL) || "1".start_with?(EVIL)
# => true
p "1".start_with?(EVIL, EVIL, EVIL)
# => false
```